### PR TITLE
Redesign Series.sort_values new-style impl

### PR DESCRIPTION
**Pull request by kozlov-alexey**

### CODE REVIEW

**Comment by kozlov-alexey** [comment]
@AlexanderKalistratov @PokhodenkoSA @densmirn 
Can you please review this PR?

File: sdc/tests/test_series.py [hunk 1]
```diff
@@ -4216,52 +4223,164 @@ def test_impl(S):
...
+            return A.sort_values(kind=literally(param_value))
+        hpat_func = self.jit(test_impl)
+
+        # using sequences of unique values because default sorting algorithm is not stable
+        n = 11
+        data_to_test = [
+            [1, -0., 0.2, -3.7, np.inf, np.nan, -1.0, 2/3, 21.2, -np.inf, 9.99],
+            np.arange(-10, 20, 1),
+            np.unique(np.random.ranf(n)),
+            np.unique(np.random.randint(0, 100, n)),
```
Comment (densmirn) [inline sdc/tests/test_series.py]: Let's set up seed before generating randoms here and in below test.

Comment (kozlov-alexey) [reply sdc/tests/test_series.py]: @densmirn Thank you, corrected.

File: sdc/datatypes/common_functions.py [hunk 2]
```diff
@@ -511,3 +513,65 @@ def _sdc_pandas_format_percentiles(arr):
...
+
+            data = A[~nan_mask]
+            keys = idx[~nan_mask]
+            if kind == 'quicksort':
+                zipped = list(zip(list(data), list(keys)))
+                zipped = quicksort_func(zipped)
+                argsorted = [zipped[i][1] for i in numpy.arange(len(data))]
+            elif kind == 'mergesort':
+                sdc.hiframes.sort.local_sort((data, ), (keys, ))
+                argsorted = list(keys)
```
[the remark this answers is not in the document]
Comment (kozlov-alexey) [reply sdc/datatypes/common_functions.py]: Done.

File: sdc/datatypes/common_functions.py [hunk 3]
```diff
@@ -511,3 +513,65 @@ def _sdc_pandas_format_percentiles(arr):
         percentiles_strs.append(p_as_string_trimmed + '%')
 
     return percentiles_strs
+
+
+def sdc_arrays_argsort(A, kind='quicksort'):
+    pass
+
+
+@overload(sdc_arrays_argsort)
```
[the remark this answers is not in the document]
Comment (kozlov-alexey) [reply sdc/datatypes/common_functions.py]: OK

**Pull request merged**